CI/samples.ci/client/petstore/java/test-manual/jersey2/ApiClientTest.java

```diff
@@ -1,249 +0,0 @@
-    public void testParameterToPairsWhenNameIsInvalid() throws Exception {
-        List<Pair> pairs_a = apiClient.parameterToPairs("csv", null, new Integer(1));
-        List<Pair> pairs_b = apiClient.parameterToPairs("csv", "", new Integer(1));
-
-        assertTrue(pairs_a.isEmpty());
-        assertTrue(pairs_b.isEmpty());
-    }
-
-    @Test
-    public void testParameterToPairsWhenValueIsNull() throws Exception {
-        List<Pair> pairs = apiClient.parameterToPairs("csv", "param-a", null);
-
-        assertTrue(pairs.isEmpty());
-    }
-
-    @Test
-    public void testParameterToPairsWhenValueIsEmptyStrings() throws Exception {
-
-        // single empty string
-        List<Pair> pairs = apiClient.parameterToPairs("csv", "param-a", " ");
-        assertEquals(1, pairs.size());
-
-        // list of empty strings
-        List<String> strs = new ArrayList<String>();
-        strs.add(" ");
-        strs.add(" ");
-        strs.add(" ");
-
-        List<Pair> concatStrings = apiClient.parameterToPairs("csv", "param-a", strs);
-
-        assertEquals(1, concatStrings.size());
-        assertFalse(concatStrings.get(0).getValue().isEmpty()); // should contain some delimiters
-    }
-
-    @Test
-    public void testParameterToPairsWhenValueIsNotCollection() throws Exception {
-        String name = "param-a";
-        Integer value = 1;
-
-        List<Pair> pairs = apiClient.parameterToPairs("csv", name, value);
-
-        assertEquals(1, pairs.size());
-        assertEquals(value, Integer.valueOf(pairs.get(0).getValue()));
-    }
-
-    @Test
-    public void testParameterToPairsWhenValueIsCollection() throws Exception {
-        Map<String, String> collectionFormatMap = new HashMap<String, String>();
-        collectionFormatMap.put("csv", ",");
-        collectionFormatMap.put("tsv", "\t");
-        collectionFormatMap.put("ssv", " ");
-        collectionFormatMap.put("pipes", "\\|");
-        collectionFormatMap.put("", ","); // no format, must default to csv
-        collectionFormatMap.put("unknown", ","); // all other formats, must default to csv
-
-        String name = "param-a";
-
-        List<Object> values = new ArrayList<Object>();
-        values.add("value-a");
-        values.add(123);
-        values.add(new Date());
-
-        // check for multi separately
-        List<Pair> multiPairs = apiClient.parameterToPairs("multi", name, values);
-        assertEquals(values.size(), multiPairs.size());
-
-        // all other formats
-        for (String collectionFormat : collectionFormatMap.keySet()) {
-            List<Pair> pairs = apiClient.parameterToPairs(collectionFormat, name, values);
-
-            assertEquals(1, pairs.size());
-
-            String delimiter = collectionFormatMap.get(collectionFormat);
-            String[] pairValueSplit = pairs.get(0).getValue().split(delimiter);
-
-            // must equal input values
-            assertEquals(values.size(), pairValueSplit.length);
-        }
-    }
-}
```

Comment:
Note: this still exists at https://github.com/OpenAPITools/openapi-generator/blob/ensures/samples/client/petstore/java/jersey2-java8/src/test/java/org/openapitools/client/ApiClientTest.java

Reply:
What are there notes for, and who are they directed at?

Reply:
They are for me and other reviewers wondering if these tests still exist

Reply:
I see. It wasn't clear if you were making yourself a note, suggesting to me that I missed a file delete, or what.

Some of our generators were deleting generated sample directories, then copying these files into the generated test directories. These files always existed in the targeted output directory, which is why GitHub doesn't show a move operation here. The delete + copy is just legacy practice which isn't really needed.

There's no reason to delete and copy these test files anymore because OpenAPI Generator doesn't overwrite existing tests and we support and honor the ignore file in the targeted directory.

Those generators which previously deleted entire samples directories and regenerated into them were probably trying to address the issue of spec files changing (either changing models, adding or removing types, etc) which could leave stale files. We've long since moved toward separating specs according to generators and only making additive changes, so much of that original sample scripting is no longer necessary.

Also as mentioned in the PR description, cleanup using the `FILES` listing will be done later. At that point, we no longer need deletes or shuffling, we can very easily clean existing samples according to what files were generated by the previous run. We may end up with a stale file occasionally causing compilation failure but I'd take that over managing 600+ files separated by OS, OAS version, and shuffling around files rather than using the ignore file functionality.

Reply:
> There's no reason to delete and copy these test files anymore because OpenAPI Generator doesn't overwrite existing tests and we support and honor the ignore file in the targeted directory.

Just a bit more background on this (come up by @jmini if my memory serves me well). The solution aims to solve the problems that sometimes users manually empty the target output folder even though the test files are not manually overwritten during code generation. We encountered this occasionally from time to time (and yes we missed it in the PR which updates a lot of files and the test case removal becomes hard to spot) and restoring the test files did help us to ensure some tests were not removed as part of the PR.

But as I pointed out before, one drawback of this method is that it's a bit counter-intuitive to the contributors that they need to update some files under the `CI` folder in order to add more tests or adjust the tests as part of the change.

I'm ok this is dropped as part of this PR and I may be able to come up with something later for us to better spot test cases removal as part of the PR review.